iothub/service/tests/Messaging/MessageClientTests.cs

```diff
@@ -0,0 +1,426 @@
+        private static IotHubServiceClientOptions s_options = new IotHubServiceClientOptions
+        {
+            Protocol = IotHubTransportProtocol.Tcp,
+            RetryPolicy = noRetryPolicy
+        };
+        private readonly RetryHandler s_retryHandler = new(new IotHubServiceNoRetry());
+
+        [TestMethod]
+        public async Task MessagesClient_OpenAsync_Cancelled_ThrowsOperationCanceledException()
+        {
+            // arrange
+            using var serviceClient = new IotHubServiceClient(
+                s_connectionString,
+                s_options);
+
+            // act
+            var ct = new CancellationToken(true);
+            Func<Task> act = async () => await serviceClient.Messages.OpenAsync(ct);
+
+            // assert
+            await act.Should().ThrowAsync<OperationCanceledException>();
+        }
+
+        [TestMethod]
+        public async Task MessagesClient_CloseAsync_Cancelled_ThrowsOperationCanceledException()
+        {
+            // arrange
+            using var serviceClient = new IotHubServiceClient(
+                s_connectionString,
+                s_options);
+
+            // act
+            var ct = new CancellationToken(true);
+            Func<Task> act = async () => await serviceClient.Messages.CloseAsync(ct);
+
+            // assert
+            await act.Should().ThrowAsync<OperationCanceledException>();
+        }
+
+        [TestMethod]
+        public async Task MessagesClient_SendAsync_WithModule_NullDeviceIdThrows()
+        {
+            // arrange
+            string payloadString = "Hello, World!";
+            byte[] payloadBytes = Encoding.UTF8.GetBytes(payloadString);
+            var mockCredentialProvider = new Mock<IotHubConnectionProperties>();
+
+            var msg = new Message(payloadBytes);
+
+            using var serviceClient = new IotHubServiceClient(
+                s_connectionString,
+                s_options);
+
+            // act
+            Func<Task> act = async () => await serviceClient.Messages.SendAsync(null, "moduleId123", msg);
+
+            // assert
+            await act.Should().ThrowAsync<ArgumentNullException>();
+        }
+
+        [TestMethod]
+        public async Task MessagesClient_SendAsync_WithModule_NullModuleIdThrows()
+        {
+            // arrange
+            string payloadString = "Hello, World!";
+            byte[] payloadBytes = Encoding.UTF8.GetBytes(payloadString);
+            var mockCredentialProvider = new Mock<IotHubConnectionProperties>();
+
+            var msg = new Message(payloadBytes);
+
+            using var serviceClient = new IotHubServiceClient(
+                s_connectionString,
+                s_options);
+
+            // act
+            Func<Task> act = async () => await serviceClient.Messages.SendAsync("deviceId123", null, msg);
+
+            // assert
+            await act.Should().ThrowAsync<ArgumentNullException>();
+        }
```

Comment:
You can use a data driven test to reduce repeated code.

```suggestion
        [TestMethod]
        [DataRow(null, "moduleId123")]
        [DataRow(string.Empty, "moduleId123")]
        [DataRow("deviceId123", null)]
        [DataRow("deviceId123", string.Empty)]
        public async Task MessagesClient_SendAsync_InvalidParamsThrows(string deviceId, string moduleId)
        {
            // arrange
            string payloadString = "Hello, World!";
            byte[] payloadBytes = Encoding.UTF8.GetBytes(payloadString);
            var mockCredentialProvider = new Mock<IotHubConnectionProperties>();

            var msg = new Message(payloadBytes);

            using var serviceClient = new IotHubServiceClient(
                s_connectionString,
                s_options);

            // act
            Func<Task> act = async () => await serviceClient.Messages.SendAsync(deviceId, moduleId, msg);

            // assert
            await act.Should().ThrowAsync<ArgumentNullException>();
        }
```